partials/CommonPublisher.java

```diff
@@ -0,0 +1,17 @@
+{% macro commonPublisher(asyncapi) %}
+package com.asyncapi.service;
+
+import org.springframework.integration.annotation.Gateway;
+import org.springframework.integration.annotation.MessagingGateway;
+
+@MessagingGateway
+public interface PublisherService {
```

Comment:
dummy question just to make sure you are using spec right, `PublisherService` is for publishing right? not subscribing? Have a look at https://www.asyncapi.com/docs/getting-started/hello-world/ publish means that you can publish a message to this channel and the app generated with your template will get this message as it is subscribed to it, does it make sense?

Reply:
It is really frustrating... Yes you are right, in this terms i mixed up subscribe and publish.
But this lead me to another idea. When we speak about message exchange it is about both parties client and server. Definition in specification state on server side, but generation for client side also make sense.
Another point to improve template, i think.

Reply:
you mean you would like to modify this template to generate both server app and client library? or you think about writing separate template?

In general yes, use case for client code generation is here too

Reply:
I think this could be done by modification of this template, because in general you should simply switch publish to subscribe and vice versa from server implementation

Reply:
if you consider client to also bo a server, no a client library

but client server... 🤔 it should have its own asyncapi file, not reuse asyncapi file of another server, right?

Reply:
Maybe client-server is bad words in case of publish and subscribe operations from asyncapi.
By specification "subscribe" operation meant that application which implement this async api allows consumers to receive defined message in the channel. For "publish" - application accepts and consumes defined message which was put to the channel by someone else. 
Is it right?
So, why not to generate this side application which could put defined message to the "publish" channel and\or consume messages which our application put into "subscribe" channel?
The entire idea based on possibility of open api tooling to generate both client and server examples based on one OpenAPI file (see [swagger editor](https://editor.swagger.io/)):
![image](https://user-images.githubusercontent.com/4137916/79245316-9e00fd00-7e80-11ea-902e-aacf55953734.png)
But maybe my understanding of principles of async api is worng and this couldn't work in this way...

Reply:
yes, I think we talk about the same, I'm just making sure that when you talk about client generation, you do not think about server, but rather a library, a client that you can use in your server, this is why I think that logically it is a separate template and not the same one

Reply:
hmmm, logically probably yes, but technically it is more or less the same code...
i'll create a new issue to discuss it

Reply:
issue #8 created from this thread